src/cert.rs

```diff
@@ -131,4 +133,63 @@ impl CertContext {
             }
         }
     }
+
+    pub fn as_chain_der_2(&self, include_root: bool, use_local_machine_chain_engine: bool) -> Result<Vec<Vec<u8>>> {
```

Comment:
What does this function do? No rustdocs and no description.

Reply:
So far just a prototype. Documentation will be added.

One option is not to include the root. The other option is to use the HKLM engine instead of default HKCU engine.

Reply:
In this case, I would rather introduce a Rust enum `ChainEngine` similar to `CertStoreType` instead of using booleans.

Reply:
Adding the different options gets a little messy.

Normally, in TLS handshake, you don't include the root. Is there any reason why we couldn't the existing as_chain_der() to exclude the root?

Reply:
Also, regarding use local_machine, we can use the CERT_ACCESS_STATE_PROP_ID to determine if HKLM instead of HKCU:
// Set if context resides in a LocalMachine SYSTEM or SYSTEM_REGISTRY store.
#define CERT_ACCESS_STATE_LM_SYSTEM_STORE_FLAG  0x4

Reply:
Maybe instead of new as_chain_der_2() update existing as_chain_der().

Reply:
will send out PR to address all the comments. will close this draft PR.